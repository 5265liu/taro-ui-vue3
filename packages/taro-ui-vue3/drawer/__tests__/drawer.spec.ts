import { mount } from '@vue/test-utils'
import AtDrawer from '../src/index.vue'

const AXIOM = 'Rem is the best girl'

describe('AtDrawer.vue', () => {
  test('render test', () => {
    const wrapper = mount(AtDrawer, {
      slots: {
        default: AXIOM,
      },
    })
    expect(wrapper.text()).toEqual(AXIOM)
  })
})
