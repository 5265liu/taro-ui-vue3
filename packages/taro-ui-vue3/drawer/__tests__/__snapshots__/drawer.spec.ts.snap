// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtDrawer Snap render AtDrawer -- props items 1`] = `
<view
  class="at-drawer at-drawer--left"
>
  <!-- mask -->
  <view
    class="at-drawer__mask"
    style="display: block; opacity: 0;"
  />
  <!-- content -->
  <view
    class="at-drawer__content"
    style="transition: all 195ms cubic-bezier(0.4, 0, 0.6, 1);"
  >
    <view
      class="at-list"
    >
      
      
      <view
        class="at-list__item at-list__item--no-border"
        data-index="0"
      >
        <view
          class="at-list__item-container"
        >
          <!-- thumb -->
          <!--v-if-->
          <!-- icon -->
          <!--v-if-->
          <!-- content -->
          <view
            class="item-content at-list__item-content"
          >
            <view
              class="item-content__info"
            >
              <view
                class="item-content__info-title"
              >
                菜单1
              </view>
              <!--v-if-->
            </view>
          </view>
          <!-- extra items -->
          <view
            class="item-extra at-list__item-extra"
          >
            <!-- extra text -->
            <!--v-if-->
            <!-- extra image -->
            <!--v-if-->
            <!-- extra switch -->
            <!--v-if-->
            <!-- extra icon - arrow -->
            <view
              class="item-extra__icon"
            >
              <view
                class="at-icon item-extra__icon-arrow at-icon-chevron-right"
              />
            </view>
          </view>
        </view>
      </view>
      <view
        class="at-list__item at-list__item--no-border"
        data-index="1"
      >
        <view
          class="at-list__item-container"
        >
          <!-- thumb -->
          <!--v-if-->
          <!-- icon -->
          <!--v-if-->
          <!-- content -->
          <view
            class="item-content at-list__item-content"
          >
            <view
              class="item-content__info"
            >
              <view
                class="item-content__info-title"
              >
                菜单2
              </view>
              <!--v-if-->
            </view>
          </view>
          <!-- extra items -->
          <view
            class="item-extra at-list__item-extra"
          >
            <!-- extra text -->
            <!--v-if-->
            <!-- extra image -->
            <!--v-if-->
            <!-- extra switch -->
            <!--v-if-->
            <!-- extra icon - arrow -->
            <view
              class="item-extra__icon"
            >
              <view
                class="at-icon item-extra__icon-arrow at-icon-chevron-right"
              />
            </view>
          </view>
        </view>
      </view>
      
      
    </view>
  </view>
</view>
`;

exports[`AtDrawer Snap render AtDrawer -- props show 1`] = `
<view
  class="at-drawer at-drawer--left"
>
  <!-- mask -->
  <view
    class="at-drawer__mask"
    style="display: block; opacity: 0;"
  />
  <!-- content -->
  <view
    class="at-drawer__content"
    style="transition: all 195ms cubic-bezier(0.4, 0, 0.6, 1);"
  >
    
    
  </view>
</view>
`;

exports[`AtDrawer Snap render AtDrawer -- props width 1`] = `
<view
  class="at-drawer at-drawer--left"
>
  <!-- mask -->
  <view
    class="at-drawer__mask"
    style="display: block; opacity: 0;"
  />
  <!-- content -->
  <view
    class="at-drawer__content"
    style="transition: all 195ms cubic-bezier(0.4, 0, 0.6, 1);"
  >
    
    
  </view>
</view>
`;

exports[`AtDrawer Snap render initial AtDrawer 1`] = `<!--v-if-->`;
